test/relayer_chain_test.go

```diff
@@ -306,6 +319,134 @@ func TestGaiaMisbehaviourMonitoring(t *testing.T) {
+	require.NoError(t, err)
+
+	// query initial balances to compare against at the end
+	var (
+		eg                       errgroup.Group
+		srcExpected, dstExpected sdk.Coins
+	)
+	eg.Go(func() error {
+		return retry.Do(func() error {
+			var err error
+			srcExpected, err = src.ChainProvider.QueryBalance(src.ChainProvider.Key())
+			if srcExpected.IsZero() {
+				return fmt.Errorf("expected non-zero balance. Err: %w", err)
+			}
+			return err
+		})
+	})
+	eg.Go(func() error {
+		return retry.Do(func() error {
+			var err error
+			dstExpected, err = dst.ChainProvider.QueryBalance(dst.ChainProvider.Key())
+			if dstExpected.IsZero() {
+				return fmt.Errorf("expected non-zero balance. Err: %w", err)
+			}
+			return err
+		})
+	})
+	require.NoError(t, eg.Wait())
+
+	// create path
+	_, err = src.CreateClients(dst, true, true, false)
+	require.NoError(t, err)
+	testClientPair(t, src, dst)
+
+	timeout, err := src.GetTimeout()
+	require.NoError(t, err)
+
+	_, err = src.CreateOpenConnections(dst, 3, timeout)
+	require.NoError(t, err)
+	testConnectionPair(t, src, dst)
+
+	_, err = src.CreateOpenChannels(dst, 3, timeout, DefaultSrcPortID, DefaultDstPortID, DefaultOrder, DefaultVersion, false)
+	require.NoError(t, err)
+
+	_, err = src.CreateOpenChannels(dst, 3, timeout, DefaultSrcPortID, DefaultDstPortID, DefaultOrder, DefaultVersion, true)
+	require.NoError(t, err)
+
+	// query open channels and ensure there are two
+	channels, err := src.ChainProvider.QueryConnectionChannels(0, src.ConnectionID())
+	require.NoError(t, err)
+	require.Equal(t, 2, len(channels))
+
+	channelOne := channels[0]
+	channelTwo := channels[1]
+
+	// send a couple of transfers to the queue on src for first channel
+	dstAddr, err := dst.ChainProvider.Address()
+	require.NoError(t, err)
+
+	srcAddr, err := src.ChainProvider.Address()
+	require.NoError(t, err)
+
+	require.NoError(t, src.SendTransferMsg(dst, testCoin, dstAddr, 0, 0, channelOne))
+	require.NoError(t, src.SendTransferMsg(dst, testCoin, dstAddr, 0, 0, channelOne))
+
+	// send a couple of transfers to the queue on dst for first channel
+	require.NoError(t, dst.SendTransferMsg(src, testCoin, srcAddr, 0, 0, channelOne))
+	require.NoError(t, dst.SendTransferMsg(src, testCoin, srcAddr, 0, 0, channelOne))
+
+	// send a couple of transfers to the queue on src for second channel
+	require.NoError(t, src.SendTransferMsg(dst, testCoin, dstAddr, 0, 0, channelTwo))
+	require.NoError(t, src.SendTransferMsg(dst, testCoin, dstAddr, 0, 0, channelTwo))
+
+	// send a couple of transfers to the queue on dst for second channel
+	require.NoError(t, dst.SendTransferMsg(src, testCoin, srcAddr, 0, 0, channelTwo))
+	require.NoError(t, dst.SendTransferMsg(src, testCoin, srcAddr, 0, 0, channelTwo))
+
+	// Wait for message inclusion in both chains
+	require.NoError(t, dst.ChainProvider.WaitForNBlocks(1))
+
+	// start the relayer process in it's own goroutine
+	rlyDone, err := relayer.StartRelayer(src, dst, 2*cmd.MB, 5)
```

Comment:
Add a `defer rlyDone()` after this line so the goroutine is stopped if we encounter a test error before the explicit call on L437.

Reply:
If I add `defer rlyDone()` but then the tests run to completion it looks like they just hang until eventually timing out which make them fail. For now, I'm just going to omit these `defer` statements.